File: experiments/adrianco/experiment-16-qwen3coder/bookshop-256k/runs/agent=qwen3-coder-local_language=typescript_prompt=ATDD/rep3/tests/test-db.js
const sqlite3 = require('sqlite3').verbose();
const path = require('path');

// Create an in-memory test database
const createTestDB = () => {
  const db = new sqlite3.Database(':memory:');
  db.serialize(() => {
    db.run(`CREATE TABLE books (
      id INTEGER PRIMARY KEY AUTOINCREMENT,
      title TEXT NOT NULL,
      author TEXT NOT NULL,
      year INTEGER,
      isbn TEXT UNIQUE
    )`);
  });
  return db;
};

// Close test database
const closeTestDB = (db) => {
  db.close();
};

module.exports = {
  createTestDB,
  closeTestDB
};

File: experiments/adrianco/experiment-64-quant-tier-30b/rest-api-crud/runs/agent=hermes-0205_language=python_model=mlxlocal/mlx-community--Qwen3-Coder-30B-A3B-Instruct-4bit_prompt=neutral_stack=q4/rep5-failed/test_db.py
#!/usr/bin/env python3
# Simple test to verify basic functionality
import sqlite3
import os

# Test database creation and basic operations
def test_database():
    # Create test database
    conn = sqlite3.connect('test_books.db')
    cursor = conn.cursor()
    
    # Create table
    cursor.execute("""
        CREATE TABLE IF NOT EXISTS books (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            title TEXT NOT NULL,
            author TEXT NOT NULL,
            year INTEGER,
            isbn TEXT UNIQUE
        )
    """)
    
    # Insert test data
    cursor.execute("INSERT INTO books (title, author, year, isbn) VALUES (?, ?, ?, ?)", 
                   ("Test Book", "Test Author", 2023, "1234567890"))
    
    conn.commit()
    
    # Query data
    cursor.execute("SELECT * FROM books")
    results = cursor.fetchall()
    
    print(f"Database test successful. Found {len(results)} books")
    conn.close()
    
    # Clean up
    os.remove('test_books.db')
    print("Database test cleaned up")

if __name__ == "__main__":
    test_database()